mllib/src/main/scala/org/apache/spark/ml/feature/SQLTransformer.scala

```diff
@@ -78,8 +79,19 @@ class SQLTransformer @Since("1.6.0") (@Since("1.6.0") override val uid: String)
   @Since("1.6.0")
   override def transformSchema(schema: StructType): StructType = {
     val spark = SparkSession.builder().getOrCreate()
-    val dummyRDD = spark.sparkContext.parallelize(Seq(Row.empty))
-    val dummyDF = spark.createDataFrame(dummyRDD, schema)
+
+    val parsedPlan = spark.sessionState.sqlParser.parsePlan($(statement))
+    parsedPlan match {
+      case _: Project =>
+      case _: Filter =>
+      case _: Aggregate =>
+      case _: Sort =>
```

Comment:
is it possible that the statement contains nested clause doing modification to data ?